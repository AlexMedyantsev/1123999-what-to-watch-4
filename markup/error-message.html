<div class="error">
  <p class="error__text"></p>
</div>