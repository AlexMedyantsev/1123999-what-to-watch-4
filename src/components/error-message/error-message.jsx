import React, {PureComponent} from "react";
import PropTypes from "prop-types";

class ErrorMessage extends PureComponent {

  render() {
    const {errorMessage, isVisible} = this.props;

    return (
      isVisible ?
        <div className="error">
          <p className="error__text">Что-то пошло не так <br></br> Ошибка: {errorMessage}</p>
        </div>
        :
        ``
    );
  }
}

ErrorMessage.propTypes = {
  errorMessage: PropTypes.number.isRequired,
  isVisible: PropTypes.bool.isRequired,
};

export {ErrorMessage};
