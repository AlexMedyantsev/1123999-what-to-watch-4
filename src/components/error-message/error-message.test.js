import React from "react";
import rerender from "react-test-renderer";
import {ErrorMessage} from "./error-message.jsx";

import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const errorMessage = `Error`;

const mockStore = configureStore([]);

describe(`Render component`, () => {
  it(`Should Error Message render correctly`, () => {
    const store = mockStore({
      CONDITION: {
        errorMessage,
      }
    });
    const tree = rerender
      .create(
          <Provider store={store}>
            <ErrorMessage
              errorMessage={errorMessage}
              isVisible={true}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
