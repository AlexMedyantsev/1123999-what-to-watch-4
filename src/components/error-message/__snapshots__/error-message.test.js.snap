// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Should Error Message render correctly 1`] = `
<div
  className="error"
>
  <p
    className="error__text"
  >
    Что-то пошло не так 
    <br />
     Ошибка: 
    Error
  </p>
</div>
`;
